/*
 * This file was automatically generated by EvoSuite
 * Mon Feb 27 08:16:01 GMT 2017
 */

package org.cs362.dominion;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.util.ArrayList;
import org.cs362.dominion.Board;
import org.cs362.dominion.GreatHall;
import org.cs362.dominion.Player;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class GreatHall_ESTest extends GreatHall_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      GreatHall greatHall0 = new GreatHall();
      ArrayList<Player> arrayList0 = new ArrayList<Player>();
      Player player0 = new Player("W;*BlI<8V.Q$I-2R0{$", false);
      Board board0 = new Board();
      player0.giveCard(greatHall0);
      greatHall0.Action(arrayList0, player0, board0);
      assertEquals(1, player0.getActions());
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      GreatHall greatHall0 = new GreatHall();
      ArrayList<Player> arrayList0 = new ArrayList<Player>();
      Board board0 = new Board();
      // Undeclared exception!
      try { 
        greatHall0.Action(arrayList0, (Player) null, board0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("org.cs362.dominion.GreatHall", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      GreatHall greatHall0 = new GreatHall();
      ArrayList<Player> arrayList0 = new ArrayList<Player>();
      Player player0 = new Player("W;*BlI<8V.Q$I-2R0{$", false);
      Board board0 = new Board();
      // Undeclared exception!
      try { 
        greatHall0.Action(arrayList0, player0, board0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }
}
